src/assets.zig

```diff
@@ -402,30 +449,78 @@ pub fn loadWorldAssets(assetFolder: []const u8, blockPalette: *Palette, biomePal
 		_ = main.models.registerModel(entry.key_ptr.*, entry.value_ptr.*);
 	}
 
-	// blocks:
 	blocks_zig.meshes.registerBlockBreakingAnimation(assetFolder);
-	for(blockPalette.palette.items) |id| {
-		const nullValue = blocks.get(id);
-		var zon: ZonElement = undefined;
-		if(nullValue) |value| {
-			zon = value;
-		} else {
-			std.log.err("Missing block: {s}. Replacing it with default block.", .{id});
-			zon = .null;
-		}
-		try registerBlock(assetFolder, id, zon);
+
+	// Blocks:
+	// First blocks from the palette to enforce ID values.
+	for(blockPalette.palette.items) |stringId| {
+		try registerBlock(assetFolder, stringId, blocks.get(stringId) orelse .null);
 	}
+
+	// Then all the blocks that were missing in palette but are present in the game.
 	var iterator = blocks.iterator();
 	while(iterator.next()) |entry| {
-		if(blocks_zig.hasRegistered(entry.key_ptr.*)) continue;
-		try registerBlock(assetFolder, entry.key_ptr.*, entry.value_ptr.*);
-		blockPalette.add(entry.key_ptr.*);
+		const stringId = entry.key_ptr.*;
+		const zon = entry.value_ptr.*;
+
+		if(blocks_zig.hasRegistered(stringId)) continue;
+
+		try registerBlock(assetFolder, stringId, zon);
+		blockPalette.add(stringId);
+	}
+
+	// Items:
+	// First from the palette to enforce ID values.
+	for(itemPalette.palette.items) |stringId| {
+		std.debug.assert(!items_zig.hasRegistered(stringId));
+
+		// Normal items should appear in items hash map.
+		if(items.get(stringId)) |zon| {
+			try registerItem(assetFolder, stringId, zon);
+			continue;
+		}
+		// But there are also items created automatically from blocks.
+		if(blocks.get(stringId)) |zon| {
+			if(!zon.get(bool, "hasItem", true)) continue;
+			try registerItem(assetFolder, stringId, zon.getChild("item"));
+			continue;
+		}
```

Comment:
How difficult would it be to handle id conflicts? (item exists in the blocks folder and in the item folder)

Reply:
What should be an outcome in such case?

Reply:
It should log an error, so the addon creator knows that they made a mistake.